### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by benjefferies**
# Motivation and Context
Initially this change was to improve the logging in
RestExceptionHandler as we were logging without adding context such as
this example ""event":"handleInvalidRequestException {}". With
https://github.com/ONSdigital/sdc/pull/4 it made sense to also look at
using those logging standards

# What has changed
* Using godaddy logger
* Improved logging details/removing unnecessary logging

# How to test?
* Run `mvn install` and check for logs from tests

```
[WARN] No valid transition; event=REQUEST_ACCEPTED; from=SUBMITTED
[INFO] Transitioning state; event=REQUEST_DISTRIBUTED; from=SUBMITTED; to=PENDING
```

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java [hunk 1]
```diff
@@ -34,8 +33,9 @@
    */
   @ExceptionHandler(CTPException.class)
   public ResponseEntity<?> handleCTPException(CTPException exception) {
-    log.error("handleCTPException");
-    log.error(exception.toString(), exception);
+    log.with("fault", exception.getFault())
+        .with("exception_message", exception.getMessage())
+        .error(exception, "Uncaught CTPException");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java]: isn't it more common to do this the other way around (i.e. message first, then the exception)?

File: src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java [hunk 2]
```diff
@@ -81,28 +80,22 @@
...
-      }
-    }
+  public ResponseEntity<?> handleInvalidRequestException(InvalidRequestException ex) {
+
+    String errors =
+        ex.getErrors()
+            .getFieldErrors()
+            .stream()
+            .map(e -> String.format("field=%s message=%s", e.getField(), e.getDefaultMessage()))
+            .collect(Collectors.joining(","));
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java]: this refactoring is a thing of beauty

File: src/main/java/uk/gov/ons/ctp/common/time/DateTimeUtil.java [hunk 3]
```diff
@@ -76,8 +78,7 @@ public static XMLGregorianCalendar stringToXMLGregorianCalendar(String string, S
       gregorianCalendar.setTime(date);
       result = DatatypeFactory.newInstance().newXMLGregorianCalendar(gregorianCalendar);
     } catch (ParseException e) {
-      log.error(String.format("%s - %s", e.getCause(), e.getMessage()));
-      log.error("Stack trace: " + e);
+      log.error(e, "Failed to parse date");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/common/time/DateTimeUtil.java]: order isn't what I'm used to seeing

File: src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java (hunk 2 above)
Comment (tej99) [reply src/main/java/uk/gov/ons/ctp/common/error/RestExceptionHandler.java]: 💯

File: pom.xml [hunk 4]
```diff
@@ -126,6 +126,12 @@
       <artifactId>jackson-datatype-jsr310</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>com.godaddy</groupId>
```
Comment (tej99) [inline pom.xml]: Worth waiting for https://github.com/ONSdigital/rm-common-config/pull/8?

File: src/main/java/uk/gov/ons/ctp/common/time/DateTimeUtil.java (hunk 3 above)
Comment (benjefferies) [reply src/main/java/uk/gov/ons/ctp/common/time/DateTimeUtil.java]: I'll sort it. I don't know why I did it that way

File: pom.xml (hunk 4 above)
Comment (benjefferies) [reply pom.xml]: Done 👍

**Pull request merged**